o.n.core/src/org/netbeans/core/NbAuthenticator.java

```diff
@@ -71,7 +74,19 @@ private NbAuthenticator() {
 
     static void install() {
         if (Boolean.valueOf(NbBundle.getMessage(GuiRunLevel.class, "USE_Authentication"))) {
-            setDefault(new NbAuthenticator());
```

Comment:
... then just call `setDefault(Lookup.getDefault().lookup(Authenticator.class))` here.

There will always be at least one instance in the lookup and the other implementations can jump in front with the `position` attribute.